packages/video_player/video_player/android/src/main/java/io/flutter/plugins/videoplayer/Messages.java

```diff
@@ -0,0 +1,446 @@
+// Autogenerated from Pigeon (v0.1.0-experimental.10), do not edit directly.
+// See also: https://pub.dev/packages/pigeon
+
+package io.flutter.plugins.videoplayer;
+
+import io.flutter.plugin.common.BasicMessageChannel;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugin.common.StandardMessageCodec;
+import java.util.HashMap;
+
+/** Generated class from Pigeon. */
+public class Messages {
+
+  /** Generated class from Pigeon that represents data sent in messages. */
+  public static class TextureMessage {
+    private Long textureId;
+
+    public Long getTextureId() {
+      return textureId;
+    }
+
+    public void setTextureId(Long setterArg) {
+      this.textureId = setterArg;
+    }
+
+    HashMap toMap() {
+      HashMap<String, Object> toMapResult = new HashMap<String, Object>();
+      toMapResult.put("textureId", textureId);
+      return toMapResult;
+    }
+
+    static TextureMessage fromMap(HashMap map) {
+      TextureMessage fromMapResult = new TextureMessage();
+      fromMapResult.textureId =
+          (map.get("textureId") instanceof Integer)
+              ? (Integer) map.get("textureId")
+              : (Long) map.get("textureId");
+      return fromMapResult;
+    }
+  }
+
+  /** Generated class from Pigeon that represents data sent in messages. */
+  public static class CreateMessage {
+    private String asset;
+
+    public String getAsset() {
+      return asset;
+    }
+
+    public void setAsset(String setterArg) {
```

Comment:
forgot if we talked about this before. Perhaps for a v1.1. Can't all these be construction arguments rather than setters? Since the "receiver" can theoretically take this object as is and just pass it around deeper into their application as immutable data. It's harder to do if it's mutable.

Reply:
Good idea I've considered it i n the past, just haven't had the time: https://github.com/flutter/flutter/issues/56496

Since java doesn't have named parameters i've never prioritized it to keep the generated code  similar across platforms.